src/Adapter/Image/Uploader/CategoryThumbnailImageUploader.php

```diff
@@ -42,50 +43,92 @@ final class CategoryThumbnailImageUploader extends AbstractImageUploader impleme
     /**
      * {@inheritdoc}
      *
-     * @throws UploadedImageConstraintException
-     * @throws ImageUploadException
+     * @throws MemoryLimitException
      * @throws ImageOptimizationException
+     * @throws ImageUploadException
+     * @throws UploadedImageConstraintException
      */
     public function upload($id, UploadedFile $uploadedImage)
     {
-        $imagesTypes = ImageType::getImagesTypes('categories');
-        $formattedName = ImageType::getFormattedName('small');
+        $this->checkImageIsAllowedForUpload($uploadedImage);
+        $this->deleteOldImage($id);
+        $this->uploadImage($id, $uploadedImage);
+        $this->generateDifferentTypes($id);
+    }
 
-        foreach ($imagesTypes as $k => $imagesType) {
-            if ($formattedName !== $imagesType['name']) {
-                continue;
-            }
+    /**
+     * Delete old category image.
+     *
+     * @param int $id
+     */
+    private function deleteOldImage($id)
```

Comment:
I'm just putting it here, but it's applicable to all files in this PR and all other PR you submitted. 

Use return types, you'll thank me later